test-framework/core/src/main/java/org/keycloak/test/framework/database/TestDatabase.java

```diff
@@ -1,109 +1,30 @@
-            }
-            databaseConfig.url(getJdbcUrl());
-            if (container instanceof MSSQLServerContainer) {
-                databaseConfig.username(((JdbcDatabaseContainer<?>) container).getUsername());
-                databaseConfig.password(((JdbcDatabaseContainer<?>) container).getPassword());
-            }
-        }
-    }
-
-    public void stop() {
-        if (databaseConfig.getContainerImage() != null) {
-            container.stop();
-            container = null;
-        } else if ("dev-mem".equals(databaseConfig.getVendor())) {
-            // TODO Stop in-mem H2 database
-        }
-    }
-
-    public Map<String, String> getServerConfig() {
-        return databaseConfig.toConfig();
-    }
-
-    public String getJdbcUrl() {
-        return ((JdbcDatabaseContainer<?>)container).getJdbcUrl();
-    }
-
-    private JdbcDatabaseContainer<?> configureJdbcContainer(JdbcDatabaseContainer<?> jdbcDatabaseContainer) {
-        if (jdbcDatabaseContainer instanceof MSSQLServerContainer) {
-            return jdbcDatabaseContainer;
-        }
-
-        return jdbcDatabaseContainer
-                .withDatabaseName("keycloak")
-                .withUsername(databaseConfig.getUsername())
-                .withPassword(databaseConfig.getPassword());
-    }
-
-    private GenericContainer<?> createContainer() {
-        return switch (databaseConfig.getVendor()) {
-            case PostgresDatabaseSupplier.VENDOR -> {
-                DockerImageName POSTGRES = DockerImageName.parse(databaseConfig.getContainerImage());
-                yield configureJdbcContainer(new PostgreSQLContainer<>(POSTGRES));
-            }
-            case MariaDBDatabaseSupplier.VENDOR -> {
-                DockerImageName MARIADB = DockerImageName.parse(databaseConfig.getContainerImage());
-                yield configureJdbcContainer(new MariaDBContainer<>(MARIADB));
-            }
-            case MySQLDatabaseSupplier.VENDOR -> {
-                DockerImageName MYSQL = DockerImageName.parse(databaseConfig.getContainerImage());
-                yield configureJdbcContainer(new MySQLContainer<>(MYSQL));
-            }
-            case MSSQLServerDatabaseSupplier.VENDOR -> {
-                DockerImageName MSSQL = DockerImageName.parse(databaseConfig.getContainerImage());
-                yield configureJdbcContainer(new MSSQLServerContainer<>(MSSQL));
-            }
-            case OracleDatabaseSupplier.VENDOR -> {
-                DockerImageName ORACLE = DockerImageName.parse(databaseConfig.getContainerImage());
-                yield configureJdbcContainer(new OracleContainer(ORACLE));
-            }
-            default -> throw new RuntimeException("Unsupported database: " + databaseConfig.getVendor());
-        };
-    }
+public interface TestDatabase {
+
+    void start();
+
+    void stop();
+
+    Map<String, String> serverConfig();
+
+//    private GenericContainer<?> createContainer() {
```

Comment:
@stianst Do you plan to keep this commented block?

Reply:
That was a mistake, removed now